### ISSUE TITLE
Advanced document access permissions doesn't work.

### ISSUE THREAD

**Issue by alexsem88** [ONLYOFFICE/onlyoffice-nextcloud#665]
I use:
* Nextcloud 24.0.2
* OnlyOffice 7.1.1-23
* Nextcloud ONLYOFFICE integration app 7.5.2

Provide advanced document access permissions using ONLYOFFICE Docs - enable:
![image](https://user-images.githubusercontent.com/52209409/179485205-86d05fda-2e41-493c-a23a-1fc76324ab0b.png)

Advanced document access permissions doesn't work:
![image](https://user-images.githubusercontent.com/52209409/179485056-6a9a3dcf-01f4-4f83-a53e-eabb23f5ed62.png)

**Comment by SergeyKorneyev** [comment]
Hello,
The advanced permissions will become available as soon as you share the document using Nextcloud's default sharing method. Our options expand the default system, they don't replace it.

**Comment by alexsem88** [comment]
The screenshot above shows that the document is shared. But there are no additional options.

**Comment by alexsem88** [comment]
Are collaborative capabilities different in Community Edition and Enterprise Edition? Extended permissions only work in Enterprise Edition or not?

**Comment by SergeyKorneyev** [comment]
The advanced permissions functionality does not depend on the product's edition. It works both with the open source and the commercial versions of Docs.

However, there are still requirements that need to be met for the options to become available. You can check the description in the first section of the following article: https://www.onlyoffice.com/blog/2022/06/nextcloud-connector_7_4_0/. For example, the Comment permission can only be enabled only if the document is shared to a user with View Only access.

I hope this clears things up.

**Comment by alexsem88** [comment]
@SergeyKorneyev 

Thanks I got it! This only works for users. For shared link it doesn't work? Are you planning to enable advanced permissions functionality for Share link?

**Comment by mageunic** [comment]
Hello, This functionality doesn't work for me ... I have shared a document with a user and the tab still to be the same as it is not shared.

Here are details on my installation : 
Nextcloud 26.0.1
Onlyoffice 7.3.3.49
Onlyoffice Nextcloud app 7.8.0

The user is allowed to modify the document on Nextcloud.

**Comment by ron-edison** [comment]
Hello, I had this functionality working on version 25, then on or around version 26 advanced permissions do not show up for newly shared files despite all requirements being met. 

Interestingly, documents shared that had advanced permissions enabled previously still work for those shares (advanced permissions tab available - review/comment permissions for read only users shared with) but newly shared files do not. Also any added read-only users shared with are not added to the advanced tab, only the original user shared with shows up.

For any newly shared documents despite being shared with a user with read-only permissions, that user does is not listed under advanced permissions, only the following is shown in the advanced tab: "Provide advanced document permissions using ONLYOFFICE Docs".

Updating to 26.0.2 does not fix this and onlyoffice docserver is the latest available - version 7.3.3-49.

Any input would be appreciated.

**Comment by alexanderonlyoffice** [comment]
Hello @ron-edison, thank you for the report, we are going to check the case and will let you know about the result.

**Comment by alexanderonlyoffice** [comment]
@ron-edison, please confirm that the issue was solved by reinstalling the connector.

**Comment by ron-edison** [comment]
Yes @alexanderonlyoffice disabling, then removing, then reinstalling the Onlyoffice connector resolved this. The various configuration settings were preserved doing this. Hopefully this helps others.

Thanks very much.

**Comment by mageunic** [comment]
> Hello, This functionality doesn't work for me ... I have shared a document with a user and the tab still to be the same as it is not shared.
> 
> Here are details on my installation : Nextcloud 26.0.1 Onlyoffice 7.3.3.49 Onlyoffice Nextcloud app 7.8.0
> 
> The user is allowed to modify the document on Nextcloud.

Hello, I still have the issue...
Currently, i'm using : 
Nextcloud 27.0.2
Onlyoffice document serveur ee : 7.4.1.36
Onlyoffice connector 8.2.0

When I share a file with another user by using nextcloud functionality, nothing appears on the advanced setting of onlyoffice....

![image](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/assets/65236802/0e00b716-d76b-4061-ba29-2a92ab103710)

![image](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/assets/65236802/4fd3963a-6377-4be9-bac9-8414f168c4f6)

**Comment by MinamiHasaki** [comment]
@mageunic The advanced sharing permission only work for internal shares, not external shares.

**Comment by mageunic** [comment]
> @mageunic The advanced sharing permission only work for internal shares, not external shares.

what do you mean by external share ? 

The document is shared to another user hosted on the same nextcloud server.... ( see my lastest post above)
This is not a public link...

**Comment by Gwylohm** [comment]
Hi, I currently use Onlyoffice connector v9.1.2 and I am not able to use the advanced document access permissions.
It does not work neither with a share with a user nor with an external share (which should now be possible with v9.1).
I uninstalled and inrestalled the connector app as mentionned by @alexanderonlyoffice and it still not works.
I don't undestand what I do wrong.
Any idea ? Any help please ? Thanks.

**Comment by cabaseira** [comment]
Same problem here.

I also receive that message, not only the users, I receive them myself, who am the administrator and owner and creator of the file.

I'm using:
- Nextcloud 29.0.0 (docker)
- Onlyoffice/documentserver-ee:6.4.2.6 (docker)
- Onlyoffice connector 9.2.0

![01](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/assets/55050120/227931c9-1906-41d4-8791-231003f3744b)
![02](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/assets/55050120/abe61788-2ba4-4949-b755-e62a325e26ec)
![03](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/assets/55050120/20b3b813-3451-4bd4-af93-c6a3c701bcd6)
![04](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/assets/55050120/e102a212-3727-4502-9ece-e575b302d914)
![05](https://github.com/ONLYOFFICE/onlyoffice-nextcloud/assets/55050120/98fafc62-090f-46dd-9e38-96469781542e)

**Comment by Jerome-Herbinet** [comment]
@LinneyS

**Comment by SergeyKorneyev** [comment]
To anyone who's having issues with the functionality, please make sure your scenario meets the conditions that are required for it to work (https://github.com/onlyoffice/onlyoffice-nextcloud?tab=readme-ov-file#advanced-document-permissions) and let us know if you have any questions or issues.

**Comment by Jerome-Herbinet** [comment]
> To anyone who's having issues with the functionality, please make sure your scenario meets the conditions that are required for it to work (https://github.com/onlyoffice/onlyoffice-nextcloud?tab=readme-ov-file#advanced-document-permissions) and let us know if you have any questions or issues.

OK @SergeyKorneyev, it works, thanks for your documentation link.

**Comment by klasyc** [comment]
> To anyone who's having issues with the functionality, please make sure your scenario meets the conditions that are required for it to work (https://github.com/onlyoffice/onlyoffice-nextcloud?tab=readme-ov-file#advanced-document-permissions) and let us know if you have any questions or issues.

Hello, I read the provided documentation, but the advanced permissions work only partially for me: When I share a spreadsheet to a Nextcloud user and give him the edit, but not the share permission (one of the edge cases stated in the documentation), I can see checkboxes for Global filter and Comment permissions. But if I give the same user the read-only access, he disappears from the Advanced tab.

Nextcloud 31.0.4, OnlyOffice connector 9.8, Document server 8.1
